### DISCUSSION TITLE
Flipper in Test Suites

### DISCUSSION THREAD

**Discussion by eyefodder** [jnunemaker/flipper#551] [Q&A] [answered]
Hey there - just started using Flipper, so apologies if this has been asked elsewhere.

I want Flipper to pretty much 'get out of the way' when I'm running my test suite. Using Rspec I started by doing this:

```ruby
config.before(:each) do
  allow(Flipper).to_receive(:enabled?).and_return(true)
end
```

and this worked, but I noticed that there's an HTTP call also being made, so I had to:
```ruby
WebMock.stub_request(:get, "https://www.flippercloud.io/adapter/features").to_return(status: 200, body: "")
```

Is there a better pattern for this? I feel like I must be missing something obvious...

**Comment by bkeepers** [comment, marked as answer]
Hi @eyefodder, you'll need to override the default flipper instance.

```ruby
config.before(:each) do
  Flipper.instance = Flipper.new(Flipper::Adapters::Memory.new)
  allow(Flipper).to_receive(:enabled?).and_return(true)
end
```

I'll work on an update to docs (or Flipper's behavior in tests) to make this more clear.

**Reply by jnunemaker** [reply]
Yep. I recommend using memory instance. 

Also, https://github.com/jnunemaker/flipper/issues/482#issuecomment-725535409 explains how we had an `EnabledByDefault` adapter at GitHub to enable all features by default in tests. We could turn it on or off based on an environment variable which was nice for matrixed test runs to ensure tests passed while features were enabled and disabled.

**Comment by jnunemaker** [comment]
> and this worked, but I noticed that there's an HTTP call also being made, so I had to:

This should/could/would only happen if you are using cloud. Are you? 

Either way, if you switch to memory adapter for tests then you won't need the web mock anymore.

**Reply by eyefodder** [reply]
Hey there - yep; this was using cloud!

**Reply by jnunemaker** [reply]
@eyefodder Also, I forgot to ask earlier, but I'd love to know what you are using flipper for as well and how it is working out for you (what went well, what was hard other than what you mentioned here).

If you aren't comfortable leaving that information here in a public comment, feel free to email me directly john@fewerandfaster.com. I'm just always curious. :)
